### PR TITLE
Added conditional check when compiling for UIKit for Mac targets

### PR DESCRIPTION
**Pull request by colinhumber**

### CODE REVIEW

File: Package.resolved [hunk 1]
```diff
@@ -6,7 +6,7 @@
         "repositoryURL": "https://github.com/mxcl/PromiseKit.git",
         "state": {
           "branch": null,
-          "revision": "e0176e7886b3bef5b9a546bcc6b357f6f569f222",
+          "revision": "fe1e9c5b62465227cceb7b0e6e79489ba7b824af",
```
Comment (mxcl) [inline Package.resolved]: This is not related, please remove.

Comment (colinhumber) [reply Package.resolved]: Sorry about that, done!

Comment (mxcl) [reply Package.resolved]: nps and thanks! Will merge and release once CI passes.
